gnome-initial-setup/pages/timezone/gis-timezone-page.c

```diff
@@ -445,6 +477,13 @@ gis_timezone_page_constructed (GObject *object)
   priv->map_location_changed_id =
     g_signal_connect (priv->map, "location-changed",
                       G_CALLBACK (map_location_changed), page);
+  g_signal_connect (priv->network_monitor, "network-changed",
```

Comment:
You'll want to disconnect this signal somewhere other than in its callback – what if the page is disposed, then the callback is called?

Reply:
I did forget to clear the `network_monitor` (I've just added that). If I do clear it on dispose, then it disconnects its signals, meaning there'll be no chance that the page is disposed and the callback is called, right?

Reply:
What if something else holds a ref to the same, shared `network_monitor`? (Why not disconnect the signal handler before clearing it?)

Reply:
Well, nothing really holds a ref to the network monitor here. But I get that it's safer, so if I can do that.